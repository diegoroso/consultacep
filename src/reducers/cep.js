const initialState = {
  data: {},
  loading: false,
  loaded: false
}

export default function reducer(state=initialState, action) {
  switch (action.type) {
    case 'FETCH_CEP':
      return {
        ...state,
        loading: true
      }

    case 'FETCH_CEP_SUCCESS':
      return {
        ...state,
        data: action.payload,
        loading: false,
        loaded: true
      }

      case 'FETCH_CEP_FAIL':
        return {
          ...state,
          data: action.payload,
          loading: false,
          loaded: true
        }

    default:
      return initialState
  }
}
