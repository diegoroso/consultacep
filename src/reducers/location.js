import {
  FETCH_CEP,
  FETCH_GEOLOCATION,
  FETCH_CEP_SUCCESS,
  FETCH_GEOLOCATION_SUCCESS,
  FETCH_CEP_FAIL,
  FETCH_GEOLOCATION_FAIL,
  RESET_TO_DEFAULT
} from '../actions/actionTypes';

const initialState = {
  data: {
    cep: '',
    logradouro: '',
    bairro: '',
    localidade: '',
    uf: '',
    coordinates: {
      lat: -14.7234029,
      lng: -51.4575307
    },
    zoom: 4,
    marker: false
  },
  error: false,
  loading: false,
  loaded: false
}

export default function reducer(state=initialState, action) {
  switch (action.type) {
    case FETCH_CEP:
    case FETCH_GEOLOCATION:
      return {
        ...state,
        loading: true,
        error: false
      }

    case FETCH_CEP_SUCCESS:
      const {cep, logradouro, bairro, localidade, uf} = action.payload
      return {
        ...state,
        data: {
          ...state.data,
          cep,
          logradouro,
          bairro,
          localidade,
          uf
        }
      }

    case FETCH_GEOLOCATION_SUCCESS:
      return {
        ...state,
        data: {
          ...state.data,
          coordinates: action.payload.location,
          zoom: action.payload.zoom,
          marker: true
        },
        loading: false,
        loaded: true
      }

    case FETCH_CEP_FAIL:
    case FETCH_GEOLOCATION_FAIL:
      return {
        ...state,
        data: initialState.data,
        error: true,
        loading: false,
        loaded: true
      }

    case RESET_TO_DEFAULT:
      return initialState

    default:
      return state
  }
}
