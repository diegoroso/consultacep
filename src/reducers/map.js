import {
  FETCH_GEOLOCATION,
  FETCH_GEOLOCATION_SUCCESS,
  FETCH_GEOLOCATION_FAIL
} from '../actions/actionTypes';

const initialState = {
  data: {
    location: {
      lat: -16.7234029,
      lng: -56.4575307
    },
    zoom: 3,
    marker: false
  },
  error: null,
  loading: false,
  loaded: false
}

export default function reducer(state=initialState, action) {
  switch (action.type) {
    case FETCH_GEOLOCATION:
      return {
        ...state,
        loading: true
      }

    case FETCH_GEOLOCATION_SUCCESS:
      return {
        ...state,
        data: {
          location: action.payload.location,
          zoom: action.payload.zoom,
          marker: true
        },
        loading: false,
        loaded: true
      }

      case FETCH_GEOLOCATION_FAIL:
        return {
          ...state,
          data: initialState.data,
          error: action.payload,
          loading: false,
          loaded: true
        }

    default:
      return state
  }
}
