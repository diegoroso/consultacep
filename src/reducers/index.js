import { combineReducers } from 'redux'

import cep from './cep';
import map from './map';

export default combineReducers({ cep, map })
