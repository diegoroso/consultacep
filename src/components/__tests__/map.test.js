import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Map from '../map'
import { initialState } from '../../reducers/location'

Enzyme.configure({ adapter: new Adapter() })

const setup = () => {
  const props = {
    location: initialState
  }

  const enzymeWrapper = shallow(<Map {...props} />)

  return {
    props,
    enzymeWrapper
  }
}

describe('Map', () => {
  const { enzymeWrapper } = setup()
  expect(enzymeWrapper).toMatchSnapshot();

  it('Verifica se o mapa esta funcionando', () => {
    const GoogleMap = shallow(enzymeWrapper.props().children)
    expect(GoogleMap).toBeTruthy()
  })
})
