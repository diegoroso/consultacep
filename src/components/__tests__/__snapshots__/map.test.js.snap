// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Map
    location={
      Object {
        "data": Object {
          "bairro": "",
          "cep": "",
          "coordinates": Object {
            "lat": -14.7234029,
            "lng": -51.4575307,
          },
          "localidade": "",
          "logradouro": "",
          "marker": false,
          "uf": "",
          "zoom": 4,
        },
        "error": false,
        "loaded": false,
        "loading": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <GoogleMap
        bootstrapURLKeys={
          Object {
            "key": "AIzaSyD2WqEZVC_PPsnICgtGQ5PPEsp07ooyISE",
          }
        }
        center={
          Object {
            "lat": -14.7234029,
            "lng": -51.4575307,
          }
        }
        debounced={true}
        distanceToMouse={[Function]}
        googleMapLoader={[Function]}
        heatmap={Object {}}
        heatmapLibrary={false}
        hoverDistance={30}
        layerTypes={Array []}
        options={[Function]}
        style={
          Object {
            "height": "100%",
            "margin": 0,
            "padding": 0,
            "position": "relative",
            "width": "100%",
          }
        }
        yesIWantToUseGoogleMapApiInternals={false}
        zoom={4}
      />,
      "className": "googlemaps",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bootstrapURLKeys": Object {
          "key": "AIzaSyD2WqEZVC_PPsnICgtGQ5PPEsp07ooyISE",
        },
        "center": Object {
          "lat": -14.7234029,
          "lng": -51.4575307,
        },
        "children": null,
        "debounced": true,
        "distanceToMouse": [Function],
        "googleMapLoader": [Function],
        "heatmap": Object {},
        "heatmapLibrary": false,
        "hoverDistance": 30,
        "layerTypes": Array [],
        "options": [Function],
        "style": Object {
          "height": "100%",
          "margin": 0,
          "padding": 0,
          "position": "relative",
          "width": "100%",
        },
        "yesIWantToUseGoogleMapApiInternals": false,
        "zoom": 4,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <GoogleMap
          bootstrapURLKeys={
            Object {
              "key": "AIzaSyD2WqEZVC_PPsnICgtGQ5PPEsp07ooyISE",
            }
          }
          center={
            Object {
              "lat": -14.7234029,
              "lng": -51.4575307,
            }
          }
          debounced={true}
          distanceToMouse={[Function]}
          googleMapLoader={[Function]}
          heatmap={Object {}}
          heatmapLibrary={false}
          hoverDistance={30}
          layerTypes={Array []}
          options={[Function]}
          style={
            Object {
              "height": "100%",
              "margin": 0,
              "padding": 0,
              "position": "relative",
              "width": "100%",
            }
          }
          yesIWantToUseGoogleMapApiInternals={false}
          zoom={4}
        />,
        "className": "googlemaps",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bootstrapURLKeys": Object {
            "key": "AIzaSyD2WqEZVC_PPsnICgtGQ5PPEsp07ooyISE",
          },
          "center": Object {
            "lat": -14.7234029,
            "lng": -51.4575307,
          },
          "children": null,
          "debounced": true,
          "distanceToMouse": [Function],
          "googleMapLoader": [Function],
          "heatmap": Object {},
          "heatmapLibrary": false,
          "hoverDistance": 30,
          "layerTypes": Array [],
          "options": [Function],
          "style": Object {
            "height": "100%",
            "margin": 0,
            "padding": 0,
            "position": "relative",
            "width": "100%",
          },
          "yesIWantToUseGoogleMapApiInternals": false,
          "zoom": 4,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
